File: zombicideandroid/src/main/java/cc/game/zombicide/android/Assignee.java
package cc.game.zombicide.android;

import cc.lib.game.GColor;
import cc.lib.utils.Reflector;
import cc.lib.zombicide.ZPlayerName;

public class Assignee extends Reflector<Assignee> implements Comparable<Assignee> {
    static {
        addAllFields(Assignee.class);
    }
    final ZPlayerName name;
    String userName = "";
    GColor color;
    boolean checked;
    @Omit
    boolean isAssingedToMe;

    @Omit
    final ZombicideActivity.CharLock lock;

    public Assignee() {
        name = null;
        lock = null;
    }

    public Assignee(ZPlayerName name, String userName, GColor color, boolean checked) {
        this.name = name;
        this.userName = userName;
        this.color = color;
        this.checked = checked;
        lock = null;
    }

    Assignee(ZombicideActivity.CharLock cl) {
        name = cl.player;
        lock = cl;
    }

    @Override
    public int compareTo(Assignee o) {
        return name.compareTo(o.name);
    }

    boolean isUnlocked() {
        return (color == null || isAssingedToMe) && lock.isUnlocked();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Assignee assignee = (Assignee) o;
        return name == assignee.name;
    }

    @Override
    public int hashCode() {
        return name.hashCode();
    }
}

File: zombicideandroid/src/main/java/cc/game/zombicide/android/Assignee.kt
package cc.game.zombicide.android

import cc.game.zombicide.android.ZombicideActivity.CharLock
import cc.game.zombicide.p2p.CommAssign
import cc.lib.reflector.Omit
import cc.lib.reflector.Reflector

class Assignee(val assignee: CommAssign) : Reflector<Assignee>(), Comparable<Assignee> {
	companion object {
		init {
			addAllFields(Assignee::class.java)
		}
	}

	@Omit
	var isAssingedToMe = false

	@Omit
	var lock: CharLock = CharLock(assignee.name, 0)

	val isClickable: Boolean
		get() = isAssingedToMe || assignee.colorId == 0

	override fun compareTo(o: Assignee): Int {
		return assignee.name.compareTo(o.assignee.name)
	}

	val isUnlocked: Boolean
		get() = (assignee.colorId == 0 || isAssingedToMe) && lock.isUnlocked

	override fun equals(o: Any?): Boolean {
		if (this === o) return true
		if (o == null || javaClass != o.javaClass) return false
		val assignee = o as Assignee
		return assignee.assignee.name === assignee.assignee.name
	}

	override fun hashCode(): Int {
		return assignee.name.hashCode()
	}
}